library/firewall_lib.py

```diff
@@ -315,6 +334,16 @@
     NM_IMPORTED = False
 
 
+# The argument to ip_interface must be a unicode string
+# Must be "cast" in python2, python3 does not need this
```

Comment:
python2 doesn't have the ipaddress module. So even with the cast this won't work. Probably better to key off HAS_IPADDRESS.